portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +27,29 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private ArrayList<String> messages;
+
+  @Override
+  public void init() {
+    messages = new ArrayList<>();
+    messages.add("It would only take one hour to drive to space.");
+    messages.add("Google's first office was a rented garage.");
+    messages.add("Walmart has lower acceptance rate than Harvard.");
+  }
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Salvador!</h1>");
+    String json = convertToJsonUsingGson(messages);
+
+    response.setContentType("application/json");
+    response.setCharacterEncoding("UTF-8");
+    response.getWriter().print(json);
+    response.getWriter().flush();
+  }
+
+  private String convertToJsonUsingGson(ArrayList<String> messages) {
+    Gson gson = new Gson();
```

Comment:
change this to a member variable, so that you don't need to create a new instance on every call.

Reply:
Ok!!